src/jit/gentree.h

```diff
@@ -120,6 +120,8 @@ enum genTreeKinds
     GTK_NOVALUE = 0x0400, // node does not produce a value
     GTK_NOTLIR  = 0x0800, // node is not allowed in LIR
 
+    GTK_NOCONTAIN = 0x1000, // this node is a value, but may not be contained
```

Comment:
> Yes but contained means that no register has been assigned to the node. So it's not clear how lowering could do the contained mem thing before RA runs.

The RA does make the final determination re: whether or not most lclVar nodes are contained, yes. In these cases lowering has typically marked the contained node as "register optional". TreeNodeInfoInit does get cute in a couple of places and preemptively mark some LCL_VAR nodes as contained if it thinks they'll never be enregistered, but in all other cases I think we use reg optional.